src/libs/hash.ts

```diff
@@ -12,23 +12,18 @@ export default function hash(args: any[]): string {
   if (!args.length) return ''
   let key = 'arg'
   for (let i = 0; i < args.length; ++i) {
-    if (args[i] === null) {
-      key += '@null'
-      continue
-    }
     let _hash
-    if (typeof args[i] !== 'object' && typeof args[i] !== 'function') {
+    if (
+      args[i] === null ||
+      (typeof args[i] !== 'object' && typeof args[i] !== 'function')
+    ) {
       // need to consider the case that args[i] is a string:
-      // args[i]        _hash
       // "undefined" -> '"undefined"'
       // undefined   -> 'undefined'
       // 123         -> '123'
       // "null"      -> '"null"'
-      if (typeof args[i] === 'string') {
-        _hash = '"' + args[i] + '"'
-      } else {
-        _hash = String(args[i])
-      }
+      // null        -> 'null'
+      _hash = JSON.stringify(args[i])
```

Comment:
`JSON.stringify(undefined)` seems returning `undefined` itself instead of a string

Reply:
True but later it will run `key += '@' + _hash` which converts it to string.

Reply:
we used to have this issue in #1075